### DISCUSSION TITLE
Problem with Auth Enable email confirmations

### DISCUSSION THREAD

**Discussion by shinokada** [supabase/supabase#5101] [Questions]
According to the [doc](https://supabase.com/docs/reference/javascript/auth-signup), I need to disable to show an error message, "User already registered".

But I'd like to enable email confirmation and show "User already registered" if someone try to register.

How can I do this?

```
If signUp() is called for an existing confirmed user:
If "Enable email confirmations" is enabled on the "Authentication" -> "Settings" page, an obfuscated / fake user object will be returned.
If "Enable email confirmations" is disabled, an error with a message "User already registered" will be returned.
```

**Comment by philippbruhin** [comment]
Did you find a workaround already? Not sure if it is the correct way, but I throw an error message as follows: https://stackoverflow.com/a/76048112/5894988

**Reply by garylake** [reply]
This will still trigger the auth email being sent though right?

I have a similar case whereby I don't need to block/warn the user but I do need to show and capture accepting terms and conditions first if it's a new user, or let the log in proceed if it's an existing user. 

My solution was to have a database function that checks the auth.users table for an account where email equals what is provided, and then if no result found, I know it's a new user and I know to show the terms. Otherwise, I do the login right away. 

Something similar could work? Your function will need security definer set search_path = public which will mean you've lost your RLS protection and you'll need to be sure your function is doing its own checks and security. In my case the function has just the email as a parameter and it only returns a boolean if the account is found or not. A bad actor could abuse this to test whether a bunch emails has an account or not but that's it...
